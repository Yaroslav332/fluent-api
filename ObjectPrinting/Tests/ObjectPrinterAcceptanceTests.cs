using ApprovalTests;
using ApprovalTests.Namers;
using ApprovalTests.Reporters;
using NUnit.Framework;

namespace ObjectPrinting.Tests
{
    [UseApprovalSubdirectory("ObjectPrinterAcceptanceTests")]
    [UseReporter(typeof(DiffReporter))]
    [TestFixture]
    public class ObjectPrinterAcceptanceTests
    {
        [Test]
        public void Demo()
        {
            var person = new Person {Name = "Alex", Age = 19};

            var printer = ObjectPrinter.For<Person>();
            Approvals.Verify(printer.PrintToString(person));
        }

        [Test]
        public void ExcludeProperty_Should_Exclude()
        {
            var person = new Person {Name = "Alex", Age = 19};
            var printer = ObjectPrinter.For<Person>()
                .ExcludeProperty(p => p.Id);
            Approvals.Verify(printer.PrintToString(person));
        }
        
        [Test]
        public void OverrideProperty_Should_OverrideCorrectly()
        {
            var person = new Person {Name = "Alex", Age = 19};
            var printer = ObjectPrinter.For<Person>()
                .ConfigForProperty(p => p.Age)
                .UseSerializeMethod(i => "Old");
            Approvals.Verify(printer.PrintToString(person));
        }
        
        [Test]
        public void OverrideProperty_Should_DiscardPrevOverrideProperty()
        {
            var person = new Person {Name = "Alex", Age = 19};
            var printer = ObjectPrinter.For<Person>()
                .ConfigForProperty(p => p.Age)
                .UseSerializeMethod(i => "Old")
                .ConfigForProperty(p => p.Age)
                .UseSerializeMethod(i => i + " Old");
            Approvals.Verify(printer.PrintToString(person));
        }
        
        [Test]
        public void OverrideProperty_ShouldNot_Work_OnExcludedProperty()
        {
            var person = new Person {Name = "Alex", Age = 19};
            var printer = ObjectPrinter.For<Person>()
                .ExcludeProperty(p => p.Age)
                .ConfigForProperty(p => p.Age)
                .UseSerializeMethod(i => "Old");
            Approvals.Verify(printer.PrintToString(person));
        }
    }
}