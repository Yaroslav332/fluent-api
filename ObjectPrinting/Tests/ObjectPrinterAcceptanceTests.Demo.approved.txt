Person
	Name = Alexander, Captain Alexander
	Birthday = 11.11.2011 0:00:00
	RunRatio = 4
	Age = 19
	RunSpeed = 20
	Parent = Person
		Id = 00000000-0000-0000-0000-000000000000
		Name = Dude, Captain Dude
		Birthday = 11.11.1990 0:00:00
		RunRatio = 2
		Age = Old
		RunSpeed = 10
		Parent = null
